Add option to generate go code only (no compile)

eroteme opened the pull request:
Resolves #24

jpkrohling commented:
I just tried it here, and here's what I see in the help text:

```
$ go run ./ --help
OpenTelemetry Collector distribution builder (dev)

Usage:
  opentelemetry-collector-builder [flags]
  opentelemetry-collector-builder [command]

Available Commands:
  help        Help about any command
  version     Version of opentelemetry-collector-builder

Flags:
      --config string            config file (default is $HOME/.otelcol-builder.yaml)
      --description string       A descriptive name for the OpenTelemetry Collector distribution (default "Custom OpenTelemetry Collector distribution")
      --generateonly             Whether builder should only generate go code with no compile of the collector
      --go string                The Go binary to use during the compilation phase (default "/usr/bin/go")
  -h, --help                     help for opentelemetry-collector-builder
      --include-core             Whether the core components should be included in the distribution (default true)
      --module string            The Go module for the new distribution (default "github.com/jpkroehling/opentelemetry-collector-builder")
      --name string              The executable name for the OpenTelemetry Collector distribution (default "otelcol-custom")
      --otelcol-version string   Which version of OpenTelemetry Collector to use as base (default "0.25.0")
      --output-path string       Where to write the resulting files (default "/tmp/otelcol-distribution549275015")
      --version string           The version for the OpenTelemetry Collector distribution (default "1.0.0")

Use "opentelemetry-collector-builder [command] --help" for more information about a command.
```

I thought Cobra would add the default value, but looks like it isn't. Sorry about that, but could you redo one of the changes you reverted? Also, might be better to have the CLI option as two words, separated by "-", to be consistent with the other options. Like: "--generate-only". As a user, I think it would be clearer to be if it's named "--skip-compilation" instead, what do you think?

eroteme commented:
If the default value is "true" corba seems to add the default value in help but not if it default to "false". I've added the comment back and changed the name of the switch

jpkrohling commented:
Thank you for your contribution!
